TIcTacToe/tic-tac-toe/src/components/Moves.js

```diff
@@ -0,0 +1,18 @@
+import React from "react";
+
+const Moves = ({ history, setXIsNext, setStepNumber }) => {
+  const jumpTo = (step) => {
+    setStepNumber(step);
+    setXIsNext(step % 2 === 0);
+  };
+
+  const _move = history.map((step, index) => (
+    <li>
+      <button onClick={() => jumpTo(index)}>
+        {index ? `Go to move # ${index}` : `Go to game start`}
+      </button>
+    </li>
+  ));
+  return <ol key={_move}>{_move}</ol>;
```

Comment:
why do you add a key to `ol` item. There's only one of them. So it's already unique, plus you should use primitives for key props not objects. Below implementation is totally fine no need to define a variable for it.


```js
<ol key={_move}>
  {history.map((step, index) => (
     <li>
       <button onClick={() => jumpTo(index)}>
         {index ? `Go to move # ${index}` : `Go to game start`}
       </button>
     </li>
  ))}
</ol>;
```